app/controllers/users/sessions_controller.rb

```diff
@@ -3,13 +3,17 @@ class Users::SessionsController < Devise::SessionsController
   private
 
     def after_sign_in_path_for(resource)
-      if !verifying_via_email? && resource.show_welcome_screen?
+      if !verifying_via_email? && (resource.show_welcome_screen? || (skip_verification? && resource.first_login?))
```

Comment:
Why are we adding logic here that should be included inside `show_welcome_screen?` ?